Source/ProceduralPart.cs

```diff
@@ -1452,16 +1452,28 @@ public void PartParentChanged(Part newParent)
 				childToParent = part.FindAttachNodeByPart(newParent);
 			}
 
+            bool srfAttached = false;
+            if (retry && childToParent == null && newParent != null)
```

Comment:
var parentIsSurfaceAttached = false;
            if (childToParent == null && newParent?.srfAttachNode?.attachedPart == part)
            {
                childToParent = newParent.srfAttachNode;
                parentIsSurfaceAttached = true;
            }
Should work and not require the ugly retry?

Reply:
Indeed that's much better. Just checked, works pretty good.